CRM/Mailing/BAO/Mailing.php

```diff
@@ -1638,21 +1638,25 @@ public static function create(&$params, $ids = array()) {
     // Create parent job if not yet created.
     // Condition on the existence of a scheduled date.
     if (!empty($params['scheduled_date']) && $params['scheduled_date'] != 'null' && empty($params['_skip_evil_bao_auto_schedule_'])) {
+      $jobParams = [
+        'mailing_id' => $mailing->id,
+        // If we are creating a new Completed mailing (e.g. import from another system) set the job to completed.
+        // Keeping former behaviour when an id is present is precautionary and may warrant reconsideration later.
+        'status' => ((empty($params['is_completed']) || !empty($params['id'])) ? 'Scheduled' : 'Complete'),
+        'is_test' => 0,
+        '_skip_evil_bao_auto_recipients_' => TRUE,
```

Comment:
@eileenmcnaughton I need to introduce a new parameter here because there need to be some flag which should prevent mailingJob::create to trigger recipient rebuild, as needed here.

Reply:
@monishdeb arg. @totten do you think we can start to grandfather to building recipients being opt-in not opt out?

Reply:
what about we go with is_create_recipients (ideally  if not set it's false, but at least a deprecated warning if not set & set in the api). I feel like we could probably find the places where MailingJob::create are currently called?

Reply:
Just a note - I checked the already merged changes & there is no case where 'mailing_id' is passed in our 'newly-calling create function' params - so we should be safe from any issues in the already-merged changes